src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -135,12 +134,17 @@ protected virtual TransactionResult Execute(Transaction tx, in BlockExecutionCon
 
             if (commit) WorldState.Commit(spec, tracer.IsTracingState ? tracer : NullTxTracer.Instance);
 
+            var gasAvailable = tx.GasLimit - intrinsicGas;
+
             // declare the execution witness to collect witness and also charge gas
             IExecutionWitness executionWitness = spec.IsVerkleTreeEipEnabled ? new VerkleExecWitness(LogManager) : new NoExecWitness();
-            executionWitness.AccessForTransaction(tx.SenderAddress!, tx.To!, !tx.Value.IsZero);
+            if (!executionWitness.AccessForTransaction(tx.SenderAddress!, tx.To!, ref gasAvailable, !tx.Value.IsZero))
+            {
+                ThrowOutOfGasException();
```

Comment:
there was no gas charge here - we just add witness here, without charging the gas

Reply:
fixed